ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/events/ACSReader.scala

```diff
@@ -392,7 +394,11 @@ private[events] object FilterTableACSReader {
       val toEnqueue =
         if (idle(task)) queueEntry(task, ids)
         else {
-          queuedRanges += (task -> queuedRanges.getOrElse(task, Vector.empty).:+(ids))
+          val previousRanges = queuedRanges.getOrElse(task, Vector.empty)
+          if (previousRanges.length >= maxQueueSize) {
+            throw new RuntimeException(s"More than $maxQueueSize id pages queued up")
```

Comment:
Could you explain when this can happen and why we need to throw here?

Reply:
We are merging multiple streams of contract ids. If one stream is for some reason faster (something we have not yet observed in practice), data would pile up in the merging operator. We prefer throwing an exception over running out of memory.